internal/files/linkedfiles.go

```diff
@@ -0,0 +1,408 @@
+// It returns true if the checksum was updated, false if it was already up-to-date.
+func (l *Link) UpdateChecksum() (bool, error) {
+	if l.UpToDate {
+		return false, nil
+	}
+	if l.IncludedFilePath == "" {
+		return false, fmt.Errorf("file path is empty for file %v", l.IncludedFilePath)
+	}
+	if l.IncludedFileContentsChecksum == "" {
+		return false, fmt.Errorf("checksum is empty for file %v", l.IncludedFilePath)
+	}
+	newContent := fmt.Sprintf("%v %v", filepath.ToSlash(l.IncludedFilePath), l.IncludedFileContentsChecksum)
+	if err := WriteFile(filepath.Join(l.WorkDir, l.LinkFilePath), []byte(newContent)); err != nil {
+		return false, fmt.Errorf("could not update checksum for file %v: %w", l.LinkFilePath, err)
+	}
+	l.LinkChecksum = l.IncludedFileContentsChecksum
+	l.UpToDate = true
+	return true, nil
+}
+
+func (l *Link) TargetFilePath(workDir ...string) string {
+	targetFilePath := filepath.FromSlash(strings.TrimSuffix(l.LinkFilePath, linkExtension))
+	wd := l.WorkDir
+	if len(workDir) > 0 {
+		wd = workDir[0]
+	}
+	return filepath.Join(wd, targetFilePath)
+}
+
+// IncludeLinkedFiles function includes linked files from the source
+// directory to the target directory.
+// It returns a slice of Link structs representing the included files.
+// It also updates the checksum of the linked files.
+// Both directories must be relative to the root.
+func IncludeLinkedFiles(fromDir, toDir string) ([]Link, error) {
+	links, err := ListLinkedFiles(fromDir)
+	if err != nil {
+		return nil, fmt.Errorf("including linked files failed: %w", err)
+	}
+	for _, l := range links {
+		if _, err := l.UpdateChecksum(); err != nil {
+			return nil, fmt.Errorf("could not update checksum for file %v: %w", l.LinkFilePath, err)
+		}
+		targetFilePath := l.TargetFilePath(toDir)
+		if err := CopyFile(
+			filepath.Join(l.WorkDir, filepath.FromSlash(l.IncludedFilePath)),
+			targetFilePath,
+		); err != nil {
+			return nil, fmt.Errorf("could not write file %v: %w", targetFilePath, err)
+		}
+	}
+
+	return links, nil
+}
+
+// ListLinkedFiles function returns a slice of Link structs representing linked files.
+func ListLinkedFiles(fromDir string) ([]Link, error) {
+	var linkFiles []string
+	if err := filepath.Walk(
+		filepath.FromSlash(fromDir),
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() && strings.HasSuffix(info.Name(), linkExtension) {
+				linkFiles = append(linkFiles, path)
+			}
+			return nil
+		}); err != nil {
+		return nil, err
+	}
+
+	links := make([]Link, len(linkFiles))
+
+	for i, f := range linkFiles {
+		l, err := NewLinkedFile(filepath.FromSlash(f))
+		if err != nil {
+			return nil, fmt.Errorf("could not initialize linked file %v: %w", f, err)
+		}
+		links[i] = l
+	}
+
+	return links, nil
+}
+
+// CopyFile function copies a file from to to inside the root.
+func CopyFile(from, to string) error {
+	from = filepath.FromSlash(from)
```

Comment:
Nit. It can be confusing to have these methods as public methods in `files`, as they seem intended to be used with paths from link files.

Maybe these methods that are using `FromSlash` should all work insted with filesystems, so they can be more easily reused. filesystems are expected to work with slashes.

Reply:
made them private, I do not think is worth making the change to use filesystems as they are and will only be used by the linked files api, if at some point we need to reuse them then we can do the changes imo